scarb/src/ops/fmt.rs

```diff
@@ -155,16 +193,13 @@ fn check_file_formatting(
 }
 
 pub trait Emittable {
-    fn emit(&self, ws: &Workspace<'_>, path: &Path, formatted: &str);
+    fn emit(&self, ws: &Workspace<'_>, formatted: &str);
 }
 
 impl Emittable for FmtEmitTarget {
-    fn emit(&self, ws: &Workspace<'_>, path: &Path, formatted: &str) {
+    fn emit(&self, ws: &Workspace<'_>, formatted: &str) {
         match self {
-            Self::Stdout => ws
-                .config()
-                .ui()
-                .print(format!("{}:\n{}", path.display(), formatted)),
+            Self::Stdout => ws.config().ui().print(format!("{}", formatted)),
```

Comment:
> I think it's just that get_formatted_file auto-inserts an empty line at EOF ?

Note that `.ui().print( {smth} )` will call `println!`. 

Not sure if this is a proper solution,  but you can override the behaviour with following: 
```
struct TextWithNewline(String);
impl Message for TextWithNewline {
    fn print_text(self)
    where
        Self: Sized,
    {
        print!("{}", self.0);
    }
}

(...)

ws.config().ui().print(TextWithNewline(format!("{}", formatted)));
```